### PR TITLE
MAINT: small improvements to the pdf backend

### PR DESCRIPTION
**Pull request by jkseppan**
## PR Summary

Improve some docstrings and comments, use generators instead of keeping `nextFoo` style counters

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_pdf.py [hunk 1]
```diff
@@ -482,21 +511,21 @@ def __init__(self, filename, metadata=None):
                          if v is not None}
 
         self.fontNames = {}     # maps filenames to internal font names
-        self.nextFont = 1       # next free internal font name
+        self.internalFontName = (Name(f'F{i}') for i in itertools.count(1))
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_pdf.py]: I don't really mind the API break (it's clearly intended to be a private API), but please add an API change note and make the new APIs private.

File: lib/matplotlib/backends/backend_pdf.py [hunk 2]
```diff
@@ -348,11 +348,22 @@ class Stream:
...
-        length of the stream, or None (to use a memory buffer); file:
-        a PdfFile; extra: a dictionary of extra key-value pairs to
-        include in the stream header; png: if the data is already
-        png compressed, the decode parameters"""
+        """
+        Parameters
+        ----------
+
+        id : int
+            object id of the stream
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pdf.py]:
```suggestion
            Object id of the stream.
```

Generally, parameter descriptions should start with a capital letter and finish with a period, even if they are not complete sentences.

(Also in the following docs).

File: lib/matplotlib/backends/backend_pdf.py [hunk 3]
```diff
@@ -424,7 +435,25 @@ class PdfFile:
...
+            `{'Creator': 'My software', 'Author': 'Me',
+            'Title': 'Awesome fig'}`
+
+            The standard keys are `'Title'`, `'Author'`, `'Subject'`,
+            `'Keywords'`, `'Creator'`, `'Producer'`, `'CreationDate'`,
+            `'ModDate'`, and `'Trapped'`. Values have been predefined
+            for `'Creator'`, `'Producer'` and `'CreationDate'`. They
+            can be removed by setting them to `None`.
+        """
+        self.objectId = itertools.count(1)  # consumed by reserveObject
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pdf.py]: Maybe better name this
```suggestion
        self._object_id_generator = itertools.count(1)  # consumed by reserveObject
```
?

Comment (jkseppan) [reply lib/matplotlib/backends/backend_pdf.py]: I gave the new names `_seq` suffixes to keep the lines slightly shorter.

**Comment by anntzer** [comment]
Not a huge deal here, but note that we normally wait for approval from two reviewers (other than the PR author) to merge (except for docs-only changes where one reviewer is enough).

**Comment by jkseppan** [comment]
Oops, sorry! I'll be less hasty in the future. -Jouni

**Comment by anntzer** [comment]
No worries :)

**Pull request merged**